diff --git a/crypto/commons/printer.c b/crypto/commons/printer.c
index 9ba87c1..5828b3a 100644
--- a/crypto/commons/printer.c
+++ b/crypto/commons/printer.c
@@ -6,7 +6,7 @@
 /*   By: iomonad <iomonad@riseup.net>               +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2019/04/10 13:29:52 by iomonad           #+#    #+#             */
-/*   Updated: 2019/04/15 14:58:46 by iomonad          ###   ########.fr       */
+/*   Updated: 2019/04/17 15:27:34 by iomonad          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -26,9 +26,9 @@ int				pprinter(t_hashing *hash,
 
 	i = 0;
 	str = (char*)hash->state;
-	if (input->method == FARG && !opts->r)
+	if (input->method == FARG && !opts->r && !opts->q)
 		ft_printf("%s (%s) = ", hash->algo, input->input);
-	else if (input->method == STRING && !opts->r)
+	else if (input->method == STRING && !opts->r && !opts->q)
 		ft_printf("%s (\"%s\") = ", hash->algo, input->input);
 	while (i < hash->dlen)
 		ft_printf("%2.2hhx", str[i++]);
@@ -48,9 +48,9 @@ int				pprinter256(t_hashing *hash,
 
 	i = 0;
 	str = (char*)hash->state;
-	if (input->method == FARG && !opts->r)
+	if (input->method == FARG && !opts->r && !opts->q)
 		ft_printf("%s (%s) = ", hash->algo, input->input);
-	else if (input->method == STRING && !opts->r)
+	else if (input->method == STRING && !opts->r && !opts->q)
 		ft_printf("%s (\"%s\") = ", hash->algo, input->input);
 	while (i < 8)
 	{
